Lib/base64.py

```diff
@@ -84,6 +93,10 @@ def b64decode(s, altchars=None, validate=False):
         s = s.translate(bytes.maketrans(altchars, b'+/'))
     if validate and not re.match(b'^[A-Za-z0-9+/]*={0,2}$', s):
         raise binascii.Error('Non-base64 digit found')
+    if not padded:
+        if validate and not re.match(b'^[A-Za-z0-9+/]*$', s):
+            raise binascii.Error('Padding found in supposedly non-padded input')
+        s += b'=='
```

Comment:
I don't understand why two ==s is always the right padding. I may be missing something.

Reply:
Probably depends on the base64 flavor of what is accepted as "right" padding, but as far as I understand the [`a2b_base64` implementation](https://github.com/python/cpython/blob/master/Modules/binascii.c#L468-L473) extra padding characters are just ignored, so always appending `==` should be safe in this context.

Reply:
I see, thanks. Maybe worth putting that in a comment in the code.

Reply:
I am using util from `django.utils.http.urlsafe_base64_encode`

https://docs.djangoproject.com/en/3.1/ref/utils/#django.utils.http.urlsafe_base64_encode